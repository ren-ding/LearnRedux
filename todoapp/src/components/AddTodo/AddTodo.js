import React from 'react';
import {StoreContext} from '../../store/store'

let nextTodoId = 0;

const AddTodo = () => {
    let input;
    return (
        <StoreContext.Consumer>
            {({store}) =>
                (<div>
                    <input ref={ node => {
                        input = node;
                    }}
                    />
                    <button onClick= {()=>{
                        store.dispatch({
                            type:'ADD_TODO',
                            id: nextTodoId++,
                            text: input.value
                        })
                        input.value='';
                    }}>
                    Add Todo
                    </button>
                 </div>
                )
            }
        
        </StoreContext.Consumer>
    );
};

export default AddTodo;