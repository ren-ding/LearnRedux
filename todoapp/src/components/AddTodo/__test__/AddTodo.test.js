import React from 'react';
import renderer from 'react-test-renderer';
import AddTodo from '../AddTodo'

describe('AddTodo',()=>{ 
    describe('render',()=>{
        it('should render match with snapshot',()=> {
            const tree = renderer.create(<AddTodo />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});