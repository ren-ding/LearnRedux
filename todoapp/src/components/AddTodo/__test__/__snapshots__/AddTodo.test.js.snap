// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo render should render match with snapshot 1`] = `
<div>
  <input />
  <button
    onClick={[Function]}
  >
    Add Todo
  </button>
</div>
`;
